### PR TITLE
[PAPPY-420] : Slack app submission feedback

### PR DESCRIPTION
**Pull request by nikhilhegde-ddw**
Ticket : https://dataworld.atlassian.net/browse/PAPPY-420

Feedback given by Slack are 
1) Please can you remove `links:read` and `links:write` as user tokens. Apps will be sent `link_shared` events for all conversation types (public, private, direct message, or multi-party direct message), regardless of your app or the installing user's channel membership so the user token scope is no longer required: https://api.slack.com/changelog/2020-11-app-unfurls-everywhere.

2) Could you please update the old Slack logo on your landing page to adhere to our updated brand guidelines (our new logo can be found here: https://brandfolder.com/slack/logos).

3) Please can you add a section to your privacy policy that states how users can request to access and delete their individual data. For deletion, we require a generalized (i.e not specific to GDPR/CCPA) statement regarding how someone can request deletion of their data. You can review our privacy policy guidelines here: https://api.slack.com/start/distributing/guidelines#privacy

4) Please can you add a section to your privacy policy that states how long personal data is retained for by your service. You can review our privacy policy guidelines here: https://api.slack.com/start/distributing/guidelines#privacy

5) In the "Data retention policy", "Data archival/removal policy", and "Data storage policy" fields, please can you explicitly state what your policies are.

6) It appears that the app may be checking whether the payload has a challenge value before using request signing verification. Instead, could you please use request signing verification before responding to the event subscriptions.

7) Please can you change "Slack integration" to "Integration for Slack" on your Landing Page.

Feedback 1, 5 are done in slack app. 

This PR address 2, 6 and 7. 

Feedback 6 required moving checking whether the payload has a [challenge value](https://github.com/datadotworld/slack-app/blob/f70bf3179c47922dd9ae2b883e7f8710ed44bb6c/server/controllers/auth.js#L102) after using [request signing verification](https://github.com/datadotworld/slack-app/blob/f70bf3179c47922dd9ae2b883e7f8710ed44bb6c/server/controllers/auth.js#L106). 

We are using the verification token currently to validate the if request came from slack. Slack allows us to use verification token now, but it will be deprecated some time in future and recommends to use signing secret for verification. 
This PR also implements signing secret verification to verify the requests from Slack.

### CODE REVIEW

File: server/controllers/auth.js [hunk 1]
```diff
@@ -98,12 +100,29 @@ const slackOauth = (req, res) => {
     });
 };
 
+const verifySlackRequest = async (headers, body) => {
+  const slackSigningSecret = process.env.SLACK_SIGNING_SECRET;
```
Comment (saltshop-boy) [inline server/controllers/auth.js:104]: Is this new process.env.SLACK_SIGNING_SECRET?

Comment (nikhilhegde-ddw) [reply server/controllers/auth.js:104]: Yes, and also verification token should be removed from app deployed in Heroku.

File: server/routes/__test__/command.test.js [hunk 2]
```diff
@@ -48,6 +49,7 @@ describe("POST /api/v1/command/ - Process slash command", () => {
   it("should handle slack ssl check properly", done => {
     const token = process.env.SLACK_VERIFICATION_TOKEN;
```
Comment (saltshop-boy) [inline server/routes/__test__/command.test.js:50]: Is this above env required

File: server/routes/__test__/command.test.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ const dwDomain = helper.DW_DOMAIN;
 describe("POST /api/v1/command/ - Process slash command", () => {
   it("should respond to slack challenge request", done => {
     const challenge = "challenge";
+    auth.verifySlackRequest = jest.fn(() => true);
```
Comment (jonaf) [inline server/routes/__test__/command.test.js:37]: do we need to add any test for the verification?

**Review by jonaf — APPROVED** [review]
lgtm

**Pull request merged**